### PR TITLE
DDF 3705 Migrate DDF-Support to Codice-Parent

### PR DESCRIPTION
**Pull request by mcalcote**
#### What does this PR do?
Migrate DDF-Support to Codice-Parent
Reorganize top level pom
Removed Hubot and Bamboo modules

#### Who is reviewing it? 
@clockard @coyotesqrl @lessarderic @pklinef @shaundmorris

#### How should this be tested? (List steps with links to updated documentation)

**PLEASE ONLY REVIEW THE FINAL COMMIT**, the rest are only present to preserve history.

#### What are the relevant tickets?
[DDF-3705 Migrate DDF-Support to Codice-Support and set up the parent pom structure in DDF.](https://codice.atlassian.net/browse/DDF-3705)

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Review by coyotesqrl — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    </description>
+    <url>https://github.com/codice/codice-parent</url>
+    <organization>
+        <name>Codice Foundation</name>
+        <url>http://www.codice.org/</url>
+    </organization>
+    <groupId>codice</groupId>
+    <artifactId>codice-parent</artifactId>
+    <scm>
+        <url>https://github.com/codice/codice-parent.git</url>
```
Comment (shaundmorris) [inline pom.xml:29]: looks like you updated everything necessary here as for as the URLs 👍

File: README.md [hunk 2]
```diff
@@ -1,7 +1,3 @@
-<<<<<<< HEAD
```
Comment (clockard) [inline README.md]: Interesting... :)

File: gitsetup/pom.xml [hunk 3]
```diff
@@ -16,20 +16,20 @@
...
     <parent>
-        <groupId>ddf.support</groupId>
-        <artifactId>support-pom</artifactId>
+        <groupId>codice</groupId>
+        <artifactId>codice-parent</artifactId>
         <version>2.3.13-SNAPSHOT</version>
     </parent>
 
-    <groupId>ddf.support</groupId>
+    <groupId>codice</groupId>
```
Comment (clockard) [inline gitsetup/pom.xml]: groupId isn't really needed since it is the same as the parent. Not really part of your change though.

**Review by clockard — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (mcalcote) [reply README.md]: 😬 I may have had some git rebasing artifacts make it into the next to last commit

**Review by shaundmorris — APPROVED** [review]
Think the true test/review will come when we try to update DDF to use this new parent file.

File: README.md [hunk 4]
```diff
@@ -1 +1,39 @@
...
+ */
+-->
+
+<img src="http://www.codice.org/images/codice-logo.png"/>
+
+# [Codice Parent \(codice-parent\)](http://www.codice.org)
+[![Build Status](https://api.travis-ci.org/codice/codice-parent.png)](https://travis-ci.org/codice/codice-parent?branch=master)
+
+
+Codice-Parent is an open source project incorporated within the DDF integration framework.
```
Comment (pklinef) [inline README.md]: ❓ Should we leave references to DDF in a Codice parent?  I was not expecting to see this in this repo.

File: gitsetup/commit-prefix.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+DDF
```
Comment (pklinef) [inline gitsetup/commit-prefix.txt]: ❗️ This will not work for non-DDF projects.

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        <connection>scm:git:https://github.com/codice/codice-parent.git</connection>
+        <developerConnection>scm:git:git@github.com:codice/codice-parent.git</developerConnection>
+    </scm>
+    <!--
+    Manually maintained and must manually update the codice.checkstyle.version property in the /parent/pom.xml to match this
+    version if you want DDF apps to use latest support-checkstyle version; and update the <version> of the <parent> in child module
+    support-checkstyle/pom.xml
+    -->
+    <version>2.3.13-SNAPSHOT</version>
```
Comment (pklinef) [inline pom.xml]: ✏️ DDF versions should not be in Codice support.

File: support-findbugs/src/main/resources/findbugs-exclude.xml [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+    <Match>
+        <Bug category="STYLE" />
+    </Match>
+    <Match>
+        <Confidence value="3" />
+    </Match>
+
+    <!-- ddf.security.Subject and has been part of several releases as part of the Security API -->
+    <Match>
+        <Class name="ddf.security.Subject"/>
```
Comment (pklinef) [inline support-findbugs/src/main/resources/findbugs-exclude.xml]: ❗️ These exclusions should be in a DDF project.

File: support-githooks/src/test/java/org/codice/git/GitIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+        source.commit().setMessage("Initial commit for source.").call();
+        // install the hooks over to the newly created git directory
+        final MockRepoHandler repHandler = new MockRepoHandler(new File(db.getWorkTree(), "gitsetup"));
+
+        repHandler.setMetadir(db.getDirectory());
+        GitHooks.install(repHandler, "");
+        // initialize a classpath.txt file since we are unit testing
+        writeTrashFile("gitsetup", "classpath.txt", System.getProperty("java.class.path"));
+        // initialize a commit-prefix.txt since we are unit testings
+        writeTrashFile("gitsetup", "commit-prefix.txt", "DDF");
```
Comment (pklinef) [inline support-githooks/src/test/java/org/codice/git/GitIntegrationTest.java]: ✏️ Would be nice to change to CODICE for the test.

File: support-owasp/src/main/resources/dependency-check-maven-config.xml [hunk 9]
```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<suppressions xmlns="https://www.owasp.org/index.php/OWASP_Dependency_Check_Suppression">
```
Comment (pklinef) [inline support-owasp/src/main/resources/dependency-check-maven-config.xml:2]: ❓ Should we move these suppression to DDF?  Can we guarantee that all of these libraries of these versions will not be used in a safe way for all Codice projects?

File: support-karaf/commands/feature-tree/src/main/java/org/codice/platform/feature/impl/FeatureTreeCommand.java [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+            "-d"})
+    private int maxDepth = 100;
+
+    @Option(name = "--no-duplicates", description = "Do not repeat feature sub-trees that have already been displayed.", aliases = {
+            "-n"})
+    private boolean noDuplicates = false;
+
+    @Option(name = "--include-repo", description = "Include only features coming from the repositories matching this regular expression.", aliases = {
+            "-i"})
+    private String repoFilter = ".*(ddf|alliance).*";
```
Comment (pklinef) [inline support-karaf/commands/feature-tree/src/main/java/org/codice/platform/feature/impl/FeatureTreeCommand.java]: ✏️ This might be an issue.

**Review by pklinef — CHANGES_REQUESTED** [review]
`ddf-intellij-code-formatter.jar` should also be renamed.

File: README.md (hunk 4 above)
Comment (mcalcote) [reply README.md]: Removed and reworded to `Codice-Parent is an open source project meant to be utilized in other Codice projects.`

File: gitsetup/commit-prefix.txt (hunk 5 above)
Comment (mcalcote) [reply gitsetup/commit-prefix.txt]: I agree, but is `CODICE` More appropriate? Do we need a prefix?

File: pom.xml (hunk 6 above)
Comment (mcalcote) [reply pom.xml]: Start at `1.0-SNAPSHOT`?

File: support-findbugs/src/main/resources/findbugs-exclude.xml (hunk 7 above)
Comment (mcalcote) [reply support-findbugs/src/main/resources/findbugs-exclude.xml]: I will remove them.

File: support-githooks/src/test/java/org/codice/git/GitIntegrationTest.java (hunk 8 above)
Comment (mcalcote) [reply support-githooks/src/test/java/org/codice/git/GitIntegrationTest.java]: updated

File: support-owasp/src/main/resources/dependency-check-maven-config.xml (hunk 9 above)
Comment (mcalcote) [reply support-owasp/src/main/resources/dependency-check-maven-config.xml:2]: Probably should be as close as possible to the areas they are suppressed in to ensure we aren't blanket covering future usages.

**Review by pklinef — APPROVED** [review]

**Comment by mcalcote** [comment]
https://travis-ci.org/mcalcote/codice-parent/builds/376532199 Travis CI Passed

**Pull request merged**